### PR TITLE
feat(subscription): ordered messages

### PR DESCRIPTION
**Pull request by callmehiphop**
This will add support for `orderingKey` in the handwritten layer.

- [x] Tests and linter pass
- [x] Code coverage does not decrease (if any source code was changed)
- [x] Appropriate docs were updated (if necessary)

### CODE REVIEW

File: src/subscriber.ts [hunk 1]
```diff
@@ -94,36 +81,51 @@ export class Message {
      * @name Message#ackId
      * @type {string}
      */
-    this.ackId = ackId;
+    this.ackId = ackId!;
```
Comment (JustinBeckwith) [inline src/subscriber.ts]: Why is the `!` needed here?

File: src/subscriber.ts [hunk 2]
```diff
@@ -94,36 +81,51 @@ export class Message {
      * @name Message#ackId
```
Comment (JustinBeckwith) [inline src/subscriber.ts]: For the constructor, can we use a real interface instead of the inline type?

**Review by JustinBeckwith — CHANGES_REQUESTED** [review]

File: src/subscriber.ts (hunk 1 above)
Comment (callmehiphop) [reply src/subscriber.ts]: I switched to use the generated types, before I just hand wrote it but I don't think we had a solution in place for that yet. As such, they mark everything as optional (even though that isn't necessarily the case)

File: src/subscriber.ts (hunk 2 above)
Comment (callmehiphop) [reply src/subscriber.ts]: Sure!

**Review by JustinBeckwith — APPROVED** [review]

**Pull request merged**